docs/_bin/generate-license-dependency-reports.py

```diff
@@ -16,62 +16,85 @@
+        print("Encountered error [{}] when generating report for {}".format(e, module_path))
+
+
+def generate_reports(druid_path, tmp_path, exclude_ext, num_threads):
+    tmp_path = os.path.abspath(tmp_path)
+    license_report_root = os.path.join(tmp_path, "license-reports")
+    license_core_path =  os.path.join(license_report_root, "core")
+    license_ext_path = os.path.join(license_report_root, "ext")
+    shutil.rmtree(license_report_root, ignore_errors=True)
+    os.makedirs(license_core_path)
+    os.makedirs(license_ext_path)
+    druid_path = os.path.abspath(druid_path)
+
+    script_args = [(druid_path, os.path.join(druid_path, "distribution", "target", "site"), license_core_path)]
+
+    if not exclude_ext:
+        extensions_core_path = os.path.join(druid_path, "extensions-core")
+        extension_dirs = os.listdir(extensions_core_path)
+        print("Found {} extensions".format(len(extension_dirs)))
+        for extension_dir in extension_dirs:
+            extension_path = os.path.join(extensions_core_path, extension_dir)
+            if not os.path.isdir(extension_path):
+                print("{} is not a directory".format(extension_path))
+                continue
+
+            extension_report_dir = "{}/{}".format(license_ext_path, extension_dir)
+            script_args.append((extension_path, os.path.join(extension_path, "target", "site"), extension_report_dir))
+    
+    print("Generating dependency reports", end="")
+    running = True
+    def print_dots():
+        while running:
+            print(".", end="", flush=True)
+            time.sleep(10)
+    dot_thread = threading.Thread(target=print_dots)
+    dot_thread.start()
 
-        extension_report_dir = "{}/{}".format(license_ext_path, extension_dir)
-        os.mkdir(extension_report_dir)
-        prev_work_dir = os.getcwd()
-        os.chdir(full_extension_dir)
+    with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as executor:
+        for module_path, report_orig_path, report_out_path in script_args:
+            executor.submit(generate_report, module_path, report_orig_path, report_out_path)
 
-        try:
-            command = "mvn -Ddependency.locations.enabled=false project-info-reports:dependencies"
-            outstr = subprocess.check_output(command, shell=True).decode('UTF-8')
-            command = "cp -r target/site {}/site".format(extension_report_dir)
-            outstr = subprocess.check_output(command, shell=True).decode('UTF-8')
-        except:
-            print("Encountered error when generating report for: " + extension_dir)
+    running = False
+    dot_thread.join()
 
-        os.chdir(prev_work_dir)
 
 if __name__ == "__main__":
     try:
         parser = argparse.ArgumentParser(description='Generating dependency reports.')
         parser.add_argument('druid_path', metavar='<Druid source path>', type=str)
         parser.add_argument('tmp_path', metavar='<Full tmp path>', type=str)
```

Comment:
We could also ask the OS for a tmp path: https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
This is not actually a tmp path, but the directory path where the dependency reports are stored. I just kept its name as it was before.